sklearn/linear_model/huber.py

```diff
@@ -0,0 +1,267 @@
+
+    if fit_intercept:
+        grad = np.zeros(n_features + 2)
+    else:
+        grad = np.zeros(n_features + 1)
+
+    # Gradient due to the squared loss.
+    X_non_outliers = -axis0_safe_slice(X, ~outliers_mask, n_non_outliers)
+    grad[:n_features] = (
+        2. / sigma * safe_sparse_dot(weighted_non_outliers, X_non_outliers))
+
+    # Gradient due to the linear loss.
+    signed_outliers = np.ones_like(outliers)
+    signed_outliers_mask = linear_loss[outliers_mask] < 0
+    signed_outliers[signed_outliers_mask] = -1.0
+    X_outliers = axis0_safe_slice(X, outliers_mask, num_outliers)
+    sw_outliers = sample_weight[outliers_mask] * signed_outliers
+    grad[:n_features] -= 2. * epsilon * (
+        safe_sparse_dot(sw_outliers, X_outliers))
+
+    # Gradient due to the penalty.
+    grad[:n_features] += alpha * 2. * w
+
+    # Gradient due to sigma.
+    grad[-1] = n_samples
+    grad[-1] -= n_sw_outliers * epsilon ** 2
+    grad[-1] -= squared_loss / sigma
+
+    # Gradient due to the intercept.
+    if fit_intercept:
+        grad[-2] = -2. * np.sum(weighted_non_outliers) / sigma
+        grad[-2] -= 2. * epsilon * np.sum(sw_outliers)
+
+    loss = n_samples * sigma + squared_loss + outlier_loss
+    loss += alpha * np.dot(w, w)
+    return loss, grad
+
+
+class HuberRegressor(LinearModel, RegressorMixin, BaseEstimator):
+    """Linear regression model that is robust to outliers.
+
+    The Huber Regressor optimizes the squared loss for the samples where
+    ``|(y - X'w) / sigma| < epsilon`` and the absolute loss for the samples
+    where ``|(y - X'w) / sigma| > epsilon``, where w and sigma are parameters
+    to be optimized. The parameter sigma makes sure that if y is scaled up
+    or down by a certain factor, one does not need to rescale epsilon to acheive
+    the same robustness.
+
+    This makes sure that the loss function is not heavily influenced by the
+    outliers while not completely ignoring their effect.
+
+    Read more in the :ref:`User Guide <huber_regression>`
+
+    Parameters
+    ----------
+    epsilon : float, greater than 1.0, default 1.35
+        The parameter epsilon controls the number of samples that should be
+        classified as outliers. The lesser the epsilon, the more robust it is
+        to outliers.
+
+    max_iter : int, default 100
+        Number of iterations that scipy.optimize.fmin_l_bfgs_b should run for.
+
+    alpha : float, default 0.0001
+        Regularization parameter.
+
+    warm_start : bool, default False
+        This is useful if the stored attributes of a previously used model
+        has to be reused. If set to False, then the coefficients will
+        be rewritten for every call to fit.
+
+    fit_intercept : bool, default True
+        Whether or not to fit the intercept. This can be set to False
+        if the data is already centered around the origin.
+
+    tol : float, default 1e-5
+        The iteration will stop when max{|proj g_i | i = 1, ..., n} <= tol
```

Comment:
How does that look in sphinx? maybe double backticks? Please check the rendering, ok? Maybe you want math?